Min # Vendingmachine step2

undervineg opened the pull request.

undervine committed:
Add sub-classes of Milk, SoftDrink and Coffee #6

Remove Menus enums from Milk, SoftDrink and Coffee classes.

undervine committed:
Add additional properties to sub-classes and override initializers #6

undervine committed:
Add addtional functions to sub-classes of Beverage #6

undervine committed:
Add more functions to sub-classes of Beverage #6

undervine committed:
Modify validation() #6

undervine committed:
Add cases to test interface functions #7

undervine committed:
Extract common properties and functions #8

undervine committed:
Override 'description' property appending class name to super.description #9

undervine committed:
Add main() to print object description #9

undervine committed:
Add README.md about Unit test and TDD

undervine committed:
Add content about using github issue/project tab

undervine committed:
Add sources

undervine committed:
Rename class name to 'BeverageTest' to separate test files

godrm commented on VendingMachine/VendingMachine/Coffee/Coffee.swift:
```diff
@@ -0,0 +1,39 @@
+    // 카페인 함유량, 뜨거운음료여부, 무가당여부, 유통기한, 칼로리
+    private let caffeineLevels: Int
+    private let isHot: Bool?
+    private let isSweetened: Bool?
```
isHot 이나 isSweetened 는 왜 Bool? 옵셔널로 지정했는지 궁금합니다.

undervineg replied:
인수가 너무 많아서 convenience 생성자를 만들어보려고 썼습니다. 그 중에서 isHot이나 isSweetened 등 일부 속성은 생성자 파라미터를 주지 않아도 자동으로 셋팅되게 하려고 옵셔널을 줬었는데요. 딱히 필요가 없게 되었네요 \ㅅ\ 지우겠습니다 ㅎㅎ

godrm commented on VendingMachine/VendingMachine/Coffee/Coffee.swift:
```diff
@@ -0,0 +1,39 @@
+
+    func isDecaffeinated() -> Bool {
+        guard caffeineLevels <= 10 else { return false }
+        return true
```
이 정도 비교식은 return (caffeineLevels <= 10) 정도가 깔끔할 것 같습니다. 
비교식을 안 쓰는게 더 좋습니다.

undervineg replied:
네 알겠습니다

godrm reviewed (commented):
아래 피드백을 확인하세요.

undervine committed:
Change optional properties to normal properties #9

Simplify 'if' statements.
Rename function 'validate()' in Beverage class to 'expired()' to get the meaning easily.

undervine committed:
Rename expired() to isExpired() #9

undervineg commented:
- 옵셔널 변수를 일반 변수로 변경
- if 문 최대한 제거
- validate() 함수명 -> isExpired()로 변경 (의미전달 쉽게)

godrm commented:
수고하셨습니다. 늦어졌지만 다음 단계에 도전하세요.
